### DISCUSSION TITLE
[UI] Global model compatibility switch

### DISCUSSION THREAD

**Discussion by Razunter** [vladmandic/sdnext#1607] [Ideas]
I think it would be good to have one major and visible global switch between incompatible things: SD 1.5 / 2.0 / Diffusers / SDXL.

So when you switch to a new mode, incompatible things get filtered out from the UI: extra networks, settings (collapsed / greyed out, probably, not hidden completely), extensions (obviously, extensions would have to support it), ControlNet models... Lots of stuff is currently incompatible with each other but displayed anyway, causing confusion and errors.

Looks like it's partially implemented already for diffusers
> Note: application will auto-disable incompatible built-in extensions when running in diffusers mode

**Comment by vladmandic** [comment]
i'd love to do that, but it would be a huge effort.

**Comment by Tillerz** [comment]
@Razunter maybe suggest that to the original a1111 branch, always worth a try. Might feature creep back in here then. :D

**Reply by vladmandic** [reply]
unless someone does a massive pr, its not happening. i wish i was wrong.
